### DISCUSSION TITLE
3.7.0 - Code cleanups and various small improvements

### DISCUSSION THREAD

**Discussion by Jacalz** [Jacalz/rymdport#171] [Announcements]
This is a medium size release with some bug fixes, code clean-ups and performance improvements. The largest change is likely all of the underlying fixes and improvements in the Fyne v2.5.x release series.

**NOTE:** This might be the last big v3.x.y release. I plan to begin working on a v4.0.0 release as a project rewrite to reduce code complexity and simplify the software to make it easier to maintain while also fix various underlying issues in the backend. This project dates back to long before I started my university education so some of the poor design choices that I made back then have been carried forward for years.

<a href='https://flathub.org/apps/details/io.github.jacalz.rymdport'><img width='200' alt='Download on Flathub' src='https://flathub.org/assets/badges/flathub-badge-en.png'/></a>

### Changelog:
- The received text is now read-only.
- Received items are now not shown until after the download starts.
- Fixed a bug where the folder icon did not display until the receive completed.
- Cleaned up and optimized various code paths.
- Bump to Go 1.21 as lowest supported compiler version.
- Updated `fyne` to [v2.5.3](https://github.com/fyne-io/fyne/releases/tag/v2.5.3).
- Updated many other dependencies for various fixes and improvements.

<hr /><em>This discussion was created from the release <a href='https://github.com/Jacalz/rymdport/releases/tag/v3.7.0'>3.7.0 - Code cleanups and various small improvements</a>.</em>

**Comment by spect95-hub** [comment]
May I ask a question? I currently using it to transfer some file due to multiple lan problem (multiple router to router) which I cannot using FTP, instead using your software, then I noticed the file is been download at receiving pc (window platform), but not directly to my set destination, a HDD, but go through the C drive which is a SSD harddisk as cache then transfer into the HDD. Due to lifespan of SSD and little space left in my C drive, would this project can cache and assemble in the directories itself like chrome download?

**Reply by Jacalz** [reply]
Of course, questions are always welcome. I'd have preferred if you opened a new discussion topic instead of an off-topic question on the release announcement but it's fine.

I suspect you mean that you transferred a directly and not a file? If I remember correctly, regular files should be downloaded directly to the destination.

**Reply by spect95-hub** [reply]
Yes, thing happens when I using folder transer.

**Reply by Jacalz** [reply]
The thing you are seeing is related to the same problem as https://github.com/Jacalz/rymdport/issues/93. If you read though the thread you'll get the majority of the details but basically it boils down to this:

Rymdport is using Magic Wormhole for transferring data between peers. Directories are implemented using a zip-file and you can't quite do a streaming decompress/unextraction as it requires reading offsets into the file. That is why it has to be stored on the file system first. 

I'm considering either just not extracting the files going forward.
